sql-to-dbsp-compiler/SQL-compiler/src/main/java/org/dbsp/sqlCompiler/compiler/frontend/ExpressionCompiler.java

```diff
@@ -624,6 +624,13 @@ node, new DBSPTypeBool(CalciteObject.EMPTY, false), DBSPOpcode.EQ,
                         return this.compilePolymorphicFunction(call, node, type,
                                 ops, 2);
                     }
+                    case "sin":
+                    case "cos":
+                    {
+                        // Turn the argument into Double
+                        ops.set(0, ops.get(0).cast(new DBSPTypeDouble(node, false)));
```

Comment:
The `false` here is not correct, it indicates that the value cannot be null.
You should instead get the current type of the operand and copy its mayBeNull value.
This also proves that there is no test for NULL as an argument, so add that too.